File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-fixed/02-non-openai-socket-with-beta-header.ts
import WebSocket from 'ws';

// Adversarial: a third-party WebSocket that happens to use a header named
// 'OpenAI-Beta' with a 'realtime=v1'-looking value, but is not actually an
// OpenAI Realtime connection. Should not be flagged.
function setupUnrelatedSocket(): WebSocket {
  const url = 'wss://internal.example.com/v1/proxy';
  return new WebSocket(url, {
    headers: {
      'OpenAI-Beta': 'realtime=v1',
    },
  });
}

File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-fixed/02_non_openai_socket_with_beta_header.py
import websockets

# Adversarial: a third-party socket reusing the "OpenAI-Beta" header name,
# but not connecting to the OpenAI Realtime endpoint.
OTHER_URL = "wss://example.com/v1/stream"


async def connect_other_socket():
    async with websockets.connect(
        OTHER_URL,
        additional_headers={"OpenAI-Beta": "realtime=v1"},
    ) as ws:
        pass
